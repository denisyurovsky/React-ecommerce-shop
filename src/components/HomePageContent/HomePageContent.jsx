import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import {
  getCategories,
  selectCategories,
} from '../../store/categories/categoriesSlice';
import {
  selectProducts,
  getHomePageProducts,
} from '../../store/products/productsSlice';
import CardsContainer from '../ProductCard/CardsContainer/CardsContainer';
import { Title } from '../Title/Title';

import { CardArea } from './CardArea/CardArea';
import { SearchArea } from './SearchArea/SearchArea';

import styles from './HomePageContent.module.scss';

export const HomePageContent = () => {
  const dispatch = useDispatch();
  const [displayedCards, setDisplayedCards] = useState([]);
  const cards = useSelector(selectProducts);
  const categories = useSelector(selectCategories);

  useEffect(() => {
    dispatch(getHomePageProducts());
    dispatch(getCategories());
  }, [dispatch]);

  useEffect(() => {
    setDisplayedCards(cards.data);
  }, [cards]);

  return (
    <section className={styles.container}>
      <Title>Welcome to Born2Die Market!</Title>
      <SearchArea
        setDisplayedCards={setDisplayedCards}
        categories={categories}
        cards={cards.data}
      />
      <CardArea>
        <CardsContainer
          products={displayedCards}
          isLoading={cards.isLoading}
          errorOccurred={cards.errorOccurred}
        />
      </CardArea>
    </section>
  );
};
