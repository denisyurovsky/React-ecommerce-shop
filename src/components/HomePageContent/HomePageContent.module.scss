@import "../../assets/styles/colors.scss";

@media all and (min-width: 1000px) {
  .container {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
  }
}

.container {
  justify-content: flex-start;
}

@media all and (max-width: 700px) {
  .header {
    font-size: 30px;
  }
}

@media all and (max-width: 500px) {
  .header {
    font-size: 25px;
  }
}
