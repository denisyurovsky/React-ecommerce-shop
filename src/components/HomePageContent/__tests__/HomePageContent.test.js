import { render, screen } from '@testing-library/react';
import React from 'react';

import { HomePageContent } from '../HomePageContent';

describe('HomePageContent component', () => {
  describe('snapshots', () => {
    it('renders a valid snapshot', () => {
      const { asFragment } = render(<HomePageContent />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
