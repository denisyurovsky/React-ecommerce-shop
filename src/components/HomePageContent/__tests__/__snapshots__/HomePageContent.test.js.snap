// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageContent component snapshots renders a valid snapshot 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <h2
      class="header"
      data-testid="title"
    >
      Welcome to Born2Die Market!
    </h2>
    <div
      class="container"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd textField css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
        data-testid="text-search-input"
      >
        <input
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          placeholder="Search by advertisement"
          type="text"
          value=""
        />
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root searchButton css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-nnbavb"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg wrapper css-1oqqzyl-MuiContainer-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
      >
        Recently added
      </p>
      <div
        class="wrapper MuiBox-root css-0"
      >
        <span
          aria-label="Products preloader"
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
